inbm-lib/inbm_common_lib/utility.py

```diff
@@ -180,11 +180,11 @@ def canonicalize_uri(url: str) -> CanonicalUri:
 
     WARNING: a string with no forward slashes will have a scheme added. e.g., 'a' -> 'https://a'
 
-    @param url: the url
+    @param url: URL
     @return canonicalized version"""
 
     if url and URL_NULL_CHAR in url:
-        raise UrlSecurityException("Unsafe characters detected in the url. Cannot process the request.")
+        raise UrlSecurityException("Unsafe characters detected in the URL. Cannot process the request.")
```

Comment:
**Suggestion:** The canonicalize_uri function should use a more robust method to check for unsafe characters in the URL to prevent potential security issues. [security]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    if url:
        if any(char in url for char in URL_UNSAFE_CHARS):
            raise UrlSecurityException("Unsafe characters detected in the URL. Cannot process the request.")
```